libminifi/src/io/InputStream.cpp

```diff
@@ -84,9 +85,9 @@ int InputStream::read(std::string &str, bool widen) {
   }
 
   std::vector<uint8_t> buffer(len);
-  uint32_t bytes_read = gsl::narrow<uint32_t>(read(buffer.data(), len));
+  const auto bytes_read = gsl::narrow<uint32_t>(read(buffer.data(), len));
   if (bytes_read != len) {
-    return -1;
+    return static_cast<size_t>(-1);
```

Comment:
as classes generally only override `read(uint8_t*, size_t)` won't using this swallow meaningful error codes coming from a base implementation? e.g. as I understand `Socket::read` might return `size_t{-2}`

Reply:
It would, but there is only one occurrence of returning -2 and I think it doesn't hit this path. That one occurrence should be cleaned up in the future anyway, so it's not an important issue IMO.
I will change it to forward the original return value, but if it mattered, we would already have an issue.

Reply:
changed to forward errors in 47358cb